api/client/cp.go

```diff
@@ -191,10 +193,29 @@ func (cli *DockerCli) copyFromContainer(srcContainer, srcPath, dstPath string, c
 		_, srcBase := archive.SplitPathDirEntry(srcInfo.Path)
 		preArchive = archive.RebaseArchiveEntries(content, srcBase, srcInfo.RebaseName)
 	}
+
+	var reader io.ReadCloser
+	if cli.isTerminalOut {
+		size := stat.Size
+		if stat.Mode.IsDir() {
+			size = stat.DSize
+		}
+		sf := streamformatter.NewStreamFormatter()
+		progressOutput := sf.NewProgressOutput(cli.out, true)
+		reader = progress.NewProgressReader(preArchive, progressOutput, size, "", "Copying")
+		defer func() {
+			// Format blank line
+			cli.out.Write(sf.FormatStatus("", ""))
```

Comment:
The newLines to NewProgressOutput means the progress bar would be printed in a new line. But this def func here is going to append a '\n' at the end of progress bar. In other words, newLines == true means the '\n' at the start of progress bar, the "def func()" here means '\n' at the end of progress bar